misc/multihash/Cargo.toml

```diff
@@ -16,6 +16,8 @@ readme = "README.md"
 documentation = "https://docs.rs/multihash/"
 
 [dependencies]
+blake2 = { version = "0.7", default-features = false }
```

Comment:
Would it be useful to make the choice of hash be a feature, so that consuming code only need to load the hash types they want?

Reply:
Yes, that would be a very good idea, I think.

Reply:
Hmm, is there a use case where a particular hash implementation will be excluded from the consumer of the library? I could imagine setting all the hash crate dependencies to be `optional= true`, but perhaps turned on by default for compatibility. Users who know ahead of time that they only want to support a particular subset of algorithms can set `default-features = false` and specify `features = ["sha2", "sha3"]`, for example, to ensure the excluded ones never deserialize successfully rather than adding extra logic to their applications to reject them when received.

Reply:
I don't really feel too strongly about this, though. If we do decide to make the hash crates optional through features, it should probably come in a separate PR as it's somewhat out of scope here, I think.